### PR TITLE
WIP: Allow passing through some headers to backends

### PR DESCRIPTION
**Pull request by jefferai**
To do:
* Fix path-help
* Reading current values on startup
* Plumb actual header values into requests
* * ^ with plugins
* * ^ with audits? Maybe we let this be covered by audited-headers

### CODE REVIEW

File: vault/logical_system.go [hunk 1]
```diff
@@ -1159,6 +1192,108 @@ func (b *SystemBackend) handlePluginReloadUpdate(req *logical.Request, d *framew
...
+
+		// Check for a nil or empty slice to have been stored
+		if len(currHeaders) == 0 {
+			if delete {
+				return nil, nil
+			}
+		}
+
+		var found bool
+		for _, v := range currHeaders {
```
Comment (vishalnayak) [inline vault/logical_system.go]: Can we use strutil.StrListContains here?

File: vault/logical_system.go [hunk 2]
```diff
@@ -1159,6 +1192,108 @@ func (b *SystemBackend) handlePluginReloadUpdate(req *logical.Request, d *framew
...
+		}
+
+		var persist bool
+		var newHeaders []string
+
+		switch {
+		case delete && found:
+			persist = true
+			newHeaders = make([]string, 0, len(currHeaders)-1)
+			for _, v := range currHeaders {
```
Comment (vishalnayak) [inline vault/logical_system.go]: Can we use strutil.StrListDelete here?

File: vault/logical_system.go [hunk 3]
```diff
@@ -1159,6 +1192,108 @@ func (b *SystemBackend) handlePluginReloadUpdate(req *logical.Request, d *framew
...
+				if v != header {
+					newHeaders = append(newHeaders, v)
+				}
+			}
+
+		case !delete && !found:
+			persist = true
+			newHeaders = make([]string, 0, len(currHeaders)+1)
```
Comment (vishalnayak) [inline vault/logical_system.go]: Doesn't the append call automatically extend the underlying array for the slice? Can we skip the `make` here?

**Comment by greysond** [comment]
Any updates on this? I'm currently writing an auth backend that needs access to a header value, so this would be wonderful.

**Comment by jefferai** [comment]
Yes, we were waiting on some internal discussion to figure out how we wanted to expose configuration for this. It should be in either 0.9.1 or 0.9.2.

**Comment by jefferai** [comment]
@greysond if you still have interest in working on this I can connect you with the person that was going to be tackling it. I don't want you two to be colliding on effort though!

**Pull request closed without merging**